### ISSUE TITLE
CSRF Reported on JavaScript page

### ISSUE THREAD

**Issue by keenam** [zaproxy/zaproxy#7890]
### Describe the bug

CSRF Alert is reported on a javascript content

JavaScript file contains the following
```js
window.test = function() [
// ...
xmlRequestFrame.document.write('<html><body><form method="post"><input type="hidden" name="__CALLBACKLOADSCRIPT" value="t"></form></body></html>');
}
```

It looks at the form element and CSRF token missing is reported.

### Steps to reproduce the behavior

1. Start Manual Explore
2. Go to a page that has a script to an element that contains `<form>` in string
 `<script src="/WebResource.axd?...">`
4. then Check the Alert
5. CSRF Alert is reported

### Expected behavior

CSRF Alert is not reported on javascript content

### Software versions

OWASP ZAP
Version: 2.12.0

Installed Add-ons: [[id=alertFilters, version=15.0.0],
[id=ascanrules, version=54.0.0], [id=automation,
version=0.28.0], [id=bruteforce, version=12.0.0],
[id=callhome, version=0.6.0], [id=commonlib,
version=1.14.0], [id=database, version=0.1.0], [id=diff,
version=12.0.0], [id=directorylistv1, version=5.0.0],
[id=domxss, version=15.0.0], [id=encoder, version=1.1.0],
[id=exim, version=0.5.0], [id=formhandler, version=6.3.0],
[id=fuzz, version=13.9.0], [id=gettingStarted,
version=14.0.0], [id=graaljs, version=0.3.0], [id=graphql,
version=0.16.0], [id=help, version=15.0.0], [id=hud,
version=0.16.0], [id=invoke, version=12.0.0], [id=network,
version=0.8.0], [id=oast, version=0.15.0], [id=onlineMenu,
version=10.0.0], [id=openapi, version=33.0.0],
[id=pscanrules, version=48.0.0], [id=quickstart,
version=37.0.0], [id=replacer, version=12.0.0], [id=reports,
version=0.20.0], [id=requester, version=7.2.0], [id=retest,
version=0.5.0], [id=retire, version=0.23.0], [id=reveal,
version=5.0.0], [id=scripts, version=38.0.0], [id=selenium,
version=15.12.1], [id=soap, version=17.0.0], [id=spider,
version=0.4.0], [id=spiderAjax, version=23.14.1], [id=tips,
version=10.0.0], [id=webdriverwindows, version=55.0.0],
[id=websocket, version=28.0.0], [id=zest, version=38.0.0]]

Operating System: Windows 10
Architecture: amd64
Java Version: AdoptOpenJDK 13.0.1
System's Locale: en_US
Display Locale: en_GB
Format Locale: en_US
Default Charset: windows-1252
ZAP Home Directory: C:\Users\knam\OWASP ZAP\
ZAP Installation Directory: C:\Program Files\OWASP\Zed Attack Proxy\.\
Look and Feel: FlatLaf Light (com.formdev.flatlaf.FlatLightLaf)

### Additional context

the javascript file content types

Content-Type: text/javascript; charset=utf-8
Content-Type: application/javascript; charset=utf-8
Content-Type: application/x-javascript

### Would you like to help fix this issue?

- [ ] Yes

**Comment by FiveOFive** [comment]
I've encountered this as well. It seems like a false positive as a form element in .js might never actually make it to the DOM or when it does it might be augemented to have anti-csrf tokens.

ZAP team - do you agree that .js files should be excluding for the absences of anti-csrf tokens? I'd be interested in fixing this one, but want to agree on a solution before starting the code.

**Comment by kingthorin** [comment]
It seems like the active CSRF rule is already skipping anything that isn't HTML (https://github.com/zaproxy/zap-extensions/blob/b4e9db92209404f5bdd9850846d2118bc508f955/addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java#L141-L144).

The passive CSRF rule (https://github.com/zaproxy/zap-extensions/blob/b4e9db92209404f5bdd9850846d2118bc508f955/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java#L88-L90) should probably do the same thing.

**Comment by psiinon** [comment]
I'm good with this rule skipping JS files too.
It _could_ find valid issues but FPs are likely to be high, and there will be better ways to check this...

**Comment by doerofthedo** [comment]
Zap also ignores the below in JavaScript page and creates false positive. Should I make a new Issue post for that?
<head>
<meta name="_csrf" content="xxxx...

**Comment by thc202** [comment]
If that's a JavaScript response it's the same issue as this one.
